railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -563,7 +563,15 @@ def prepare!
       end
 
       def self.default_rc_file
-        File.expand_path("~/.railsrc")
+        return File.expand_path("~/.railsrc") unless ENV["XDG_CONFIG_HOME"]
+
+        xdg_railsrc = ENV["XDG_CONFIG_HOME"] + "/rails/railsrc"
+        rc_file = if File.exist?(xdg_railsrc)
+          xdg_railsrc
+        else
+          "~/.railsrc"
+        end
+        File.expand_path(rc_file)
```

Comment:
I feel like this control flow is a little hard to follow.  What do you think about this arrangment:

```suggestion
        xdg_railsrc = File.expand_path("rails/railsrc", ENV["XDG_CONFIG_HOME"]) if ENV["XDG_CONFIG_HOME"]

        if xdg_railsrc && File.exist?(xdg_railsrc)
          xdg_railsrc
        else
          File.expand_path("~/.railsrc")
        end
```

Reply:
@jonathanhefner Thanks!

Yep, I like the the flow and readability of this better, I've updated the commits now.  

I've updated the USAGE file as well. 

Hmm, I did a search for ~/.railsrc so I missed the File.join("~", ".railsrc"), good catch!

One option might be to extract the xdg detection logic into a shared parent module perhaps? 

We could do something like Rails.rc_file but I'm guessing the Rails module itself is fairly slow to change?

Once we decide on the location for storing the logic I'll attempt to write a test for it. 

Thanks for getting back to me. This is actually my first Open Source PR. 🙂